File: Which section did you scroll to?.dart
int getSectionId(int n, List<int> a) {
  int start = 0;
  for(int i=0;i<a.length;i++){
    final end = start + a[i];
    if(n >= start && n < end)return i;
    start = end;
  }
  return -1;
}

File: Which section did you scroll to?/Which section did you scroll to?.ts
export function getSectionId(scroll: number, sizes: number[]) {
  let start = 0;
  for (let i = 0; i < sizes.length; i++) {
    const end = start + sizes[i];
    if (scroll >= start && scroll < end) return i;
    start = end;
  }
  return -1;
}
